pkg/api/osa/vlabs/types.go

```diff
@@ -0,0 +1,99 @@
+package vlabs
+
+// OpenShiftCluster complies with the ARM model of resource definition in a JSON
+// template.
+type OpenShiftCluster struct {
+	ID         string                `json:"id,omitempty"`
+	Location   string                `json:"location,omitempty"`
+	Name       string                `json:"name,omitempty"`
+	Plan       *ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags       map[string]string     `json:"tags,omitempty"`
+	Type       string                `json:"type,omitempty"`
+	Properties Properties            `json:"properties,omitempty"`
+}
+
+// ResourcePurchasePlan defines the resource plan as required by ARM for billing
+// purposes.
+type ResourcePurchasePlan struct {
+	Name          string `json:"name,omitempty"`
+	Product       string `json:"product,omitempty"`
+	PromotionCode string `json:"promotionCode,omitempty"`
+	Publisher     string `json:"publisher,omitempty"`
+}
+
+// Properties represents the cluster definition.
+type Properties struct {
+	ProvisioningState ProvisioningState `json:"provisioningState,omitempty"`
+	OpenShiftVersion  string            `json:"openShiftVersion,omitempty"`
+	// TODO: need to clarify external API for PublicHostname and decide what to
+	// implement when.  Allow users to specify a PublicHostname then have them
+	// create a CNAME to a FQDN we return?  Allow users to not specify and we
+	// return a FQDN?  In which case, how will the plugin know the FQDN?
+	PublicHostname string `json:"publicHostname,omitempty"`
+	// FQDN string `json:"fqdn,omitempty"` // TODO: do we need to add this?
+	// TODO: need to clarify external API for RoutingConfigSubdomain.  Do we
+	// create one and return it if it's not provided?  Will this be transparent
+	// to the plugin?
+	RoutingConfigSubdomain string `json:"routingConfigSubdomain,omitempty"`
+	// TODO: need to clarify the external API for AgentPoolProfiles.  Will we
+	// allow users to specify an `infra` pool?
```

Comment:
My understanding is infra pool/VMs is definitely need for OS. In that case customer shouldn't have choice and should just get one (plus it can be named appropriately to additional context).

Reply:
I think at a minimum we'd allow users to select some options of the infra pool.  Specifically for sizing.  Maybe they don't need all of the setting exposed there.